MySQLdb/_mysql.c

```diff
@@ -522,7 +539,12 @@ _mysql_ConnectionObject_Initialize(
     }
 #ifdef HAVE_ENUM_MYSQL_OPT_SSL_MODE
     if (ssl_mode) {
-        int ssl_mode_num = _get_ssl_mode_num(ssl_mode);
+        char *corrected_ssl_mode = NULL;
+        if (strcasecmp(ssl_mode, "REQUIRED") == 0 || strcasecmp(ssl_mode, "VERIFY_CA") == 0)
+            corrected_ssl_mode = "VERIFY_IDENTITY";
```

Comment:
Yes, it was added for consistency sake, in order to keep the same behavior for MySql connectors and MariaDB connectors.

Reply:
So you're right that `VERIFY_CA` can't be expressed easily in MariaDB, but `REQUIRED` can. So going also back to my comment above, `REQUIRED` in both MySQL and MariaDB can be only enforcing TLS and not validating the certificate? 

Matching behavior would be the least confusing? I think for `VERIFY_CA` it's debatable if that should then map to the MySQL equivalent of `VERIFY_IDENTITY` or `REQUIRED`, could argue for both really.

Reply:
Yes, MariaDB offers less granularity and right now I don't see how to express VERIFY_CA via it, going to check it. maybe something has changed. REQUIRED in MariaDB is unfortunately different than in MySQL, since it does not prevent from falling back to non-TLS connection if TLS was not established. So that is not an option as well. 
So we can have something like this:
REQUIRED in MySQL works as it was working before.
REQUIRED in MariaDB is failing back to non-TLS if TLS is not established(since it is the way the MariaDB is implemented)
VERIFY_CA in MySQL works the same as it was before
VERIFY_CA in MariaDB - works the same as in VERIFY_INDETITY since MariaDB does not offer such granularity.
VERIFY_IDENTITY in MySQL works the same as it was before
VERIFY_IDENTITY mimics the MySQL behaviour via MYSQL_OPT_SSL_VERIFY_SERVER_CERT

If this is fine with you I will make the necessary changes.

Reply:
@thereisnosun Yep, that's my understanding as well indeed. Fwiw, https://github.com/brianmario/mysql2/pull/1205 does something similar for the Ruby MySQL driver where `VERIFY_CA` is not available when built against MariaDB and the other options are mapped onto how MariaDB behaves.